### PR TITLE
Padimitrov/dsc resource pool

### PR DESCRIPTION
**Pull request by paul-dimitrov**
Added Resource Pool DSC resource.

### CODE REVIEW

File: Source/VMware.vSphereDSC/Classes/InventoryUtil.ps1 [hunk 1]
```diff
@@ -45,6 +45,8 @@ class InventoryUtil {
     hidden [string] $CouldNotFindFolderMessage = "Could not find Folder {0} located in Folder {1}."
     hidden [string] $CouldNotFindInventoryItemMessage = "Could not find Inventory Item {0} located in Inventory Item {1}."
     hidden [string] $CouldNotFindDatastoreClusterMessage = "Could not find Datastore Cluster {0} located in Folder {1}."
+    hidden [string] $CouldNotFindResourcePool = "Could not find ResourcePool {0}. For more information {1}."
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Classes/InventoryUtil.ps1]: You can remove the **`.`** at the end, as the **`error message`** already contains a **`.`** at the end.

File: Source/VMware.vSphereDSC/Classes/InventoryUtil.ps1 [hunk 2]
```diff
@@ -426,4 +428,51 @@ class InventoryUtil {
...
         }
     }
+
+    <#
+    .DESCRIPTION
+
+    Retrieves the Resource pool location
+    #>
+    [PSObject] GetResourcePoolParent($resourcePoolLocation){
+        $locationItems = $resourcePoolLocation.Split('/')
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/Classes/InventoryUtil.ps1:442]: You can add here additional checks for empty location as well as location not containing **`/`**. In the first case, you should return the root folder for **`ResourcePool Inventory Items`** and in the second, call **`Get-Inventory`** with the **`$resourcePoolLocation`** parameter. 

For reference, you can check how **`GetDatacenter`** function is implemented here in the **`InventoryUtil`**.

File: Source/VMware.vSphereDSC/DSCResources/ResourcePool.ps1 [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+
+Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+
+Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#>
+
+[DscResource()]
+class ResourcePool : BaseDSC {
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/DSCResources/ResourcePool.ps1]: You can inherit **`InventoryBaseDSC`** class which already exposes **`Name, Location and Ensure`** properties. This way you can remove from here **`ResourcePoolName, ResourcePoolLocation and Ensure`**.

File: Source/VMware.vSphereDSC/DSCResources/ResourcePool.ps1 [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+       'Low' = 81920
+       'Normal' = 163840
+       'High' = 327680
+    }
+
+    [void] Set() {
+        try {
+            $this.ConnectVIServer()
+            $this.InitInventoryUtil()
+            $resourcePool = $this.InventoryUtil.GetResourcePool($this.ResourcePoolName, $this.ResourcePoolLocation)
```
Comment (SimeonGerginov) [inline Source/VMware.vSphereDSC/DSCResources/ResourcePool.ps1]: You can change the **`GetResourcePool`** method signature to accept the parent directly. This way you don't have to call the **`GetResourcePoolParent`** more than once per method. Currently if you create a new **`ResourcePool`**, you call it twice, once to check if it exists and another when you're creating it.

**Review by SimeonGerginov — CHANGES_REQUESTED** [review]

File: Source/VMware.vSphereDSC/Classes/InventoryUtil.ps1 (hunk 1 above)
Comment (paul-dimitrov) [reply Source/VMware.vSphereDSC/Classes/InventoryUtil.ps1]: Dot at the end removed.

File: Source/VMware.vSphereDSC/Classes/InventoryUtil.ps1 (hunk 2 above)
Comment (paul-dimitrov) [reply Source/VMware.vSphereDSC/Classes/InventoryUtil.ps1:442]: If the location is empty string, I'll throw an exception. As there's no single root resource pool. There's a 'Resources' RP for each host outside cluster and each cluster. 
If there is no '/' symbol in the location the code already does what you suggest. Splitting by non-existing symbol results one element array, containing the same string:
```
$a = "NoDash"
$b = $a.Split('/')
$b[0]
NoDash
```
Please correct me if I miss something.

File: Source/VMware.vSphereDSC/DSCResources/ResourcePool.ps1 (hunk 4 above)
Comment (paul-dimitrov) [reply Source/VMware.vSphereDSC/DSCResources/ResourcePool.ps1]: Fixed

File: Source/VMware.vSphereDSC/DSCResources/ResourcePool.ps1 (hunk 3 above)
Comment (paul-dimitrov) [reply Source/VMware.vSphereDSC/DSCResources/ResourcePool.ps1]: InventoryBaseDSC inherited

**Review by SimeonGerginov — APPROVED** [review]
LGTM

**Pull request merged**